src/twisted/application/runner/_pidfile.py

```diff
@@ -0,0 +1,202 @@
+# -*- test-case-name: twisted.application.runner.test.test_pidfile -*-
+# Copyright (c) Twisted Matrix Laboratories.
+# See LICENSE for details.
+
+"""
+PID file.
+"""
+
+import errno
+from os import getpid, kill
+
+from twisted.python.filepath import IFilePath
+from twisted.logger import Logger
+
+
+
+class PIDFile(object):
+    """
+    PID file.
+
+    Manages a file that remembers a process ID.
+    """
+
+    _log = Logger()
+
+
+    @staticmethod
+    def format(pid=None):
+        """
+        Format a PID file's content.
+
+        @param pid: A process ID.
+        @type pid: int
+
+        @return: Formatted PID file contents.
+        @rtype: L{bytes}
+        """
+        if pid is not None:
+            return u"{}\n".format(pid).encode("utf-8")
+        else:
+            return b""
+
+
+    def __init__(self, filePath):
+        """
+        @param filePath: The path to the PID file on disk.
+        @type filePath: L{IFilePath}
+        """
+        self.filePath = IFilePath(filePath)
+
+
+    def read(self):
+        """
+        Read the process ID stored in this PID file.
+
+        @return: The contained process ID, or C{None} if the PID file does not
```

Comment:
Agreed.  I think I like custom, so we're very explicit.

 * 5619da41be4a5f21a67c76dcdac25409e8561446  - Adds InvalidPIDFileError custom exception for the cases where the PID file has invalid contents.
 * 71781a7991f38faca4d26e6564ecc6c73ebd0067 - Adds NoPIDFound custom exception for when there is no PID (non-existing PID file).